### PR TITLE
[system-health] Add fan direction check for system health

### PR DESCRIPTION
**Pull request by Junchao-Mellanox**
#### Why I did it

Add fan direction check to system health, all fans should be in the same direction

#### How I did it

Add fan direction check to system health, all fans should be in the same direction

#### How to verify it

Manual test
Unit test
Added sonic-mgmt test case to verify

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211

### CODE REVIEW

**Review by keboliu — DISMISSED** [review]

**Comment by Junchao-Mellanox** [comment]
Hi @qiluo-msft , @prgeor , could you please help review this?

File: src/system-health/tests/test_system_health.py [hunk 1]
```diff
@@ -298,14 +298,16 @@ def test_hardware_checker():
...
+            'direction': 'intake'
         },
         'FAN_INFO|fan2': {
             'presence': 'False',
             'status': 'True',
             'speed': '60',
             'speed_target': '60',
-            'speed_tolerance': '20'
+            'speed_tolerance': '20',
+
```
Comment (liat-grozovik) [inline src/system-health/tests/test_system_health.py:310]: why is this extra line is needed?

**Comment by liat-grozovik** [comment]
@prgeor your comments are very welcome. can you please review?

File: src/system-health/health_checker/hardware_checker.py [hunk 2]
```diff
@@ -82,6 +82,7 @@ def _check_fan_status(self, config):
             self.set_object_not_ok('Fan', 'Fan', 'Failed to get fan information')
             return
```
Comment (prgeor) [inline src/system-health/health_checker/hardware_checker.py:84]: @Junchao-Mellanox please update the description of the _check_fan_status() to include fan direction check

File: src/system-health/health_checker/hardware_checker.py [hunk 3]
```diff
@@ -133,6 +134,18 @@ def _check_fan_status(self, config):
...
                         continue
 
+            if not self._ignore_check(config.ignore_devices, 'fan', name, 'direction'):
+                direction = data_dict.get('direction', 'N/A')
+                # ignore fan whose direction is not available to avoid too many false alarms
+                if direction != 'N/A':
+                    if not expect_fan_direction:
+                        # initialize the expect fan direction
+                        expect_fan_direction = (name, direction)
+                    elif direction != expect_fan_direction[1]:
```
Comment (prgeor) [inline src/system-health/health_checker/hardware_checker.py:144]: @Junchao-Mellanox I don't understand why `expect_fan_direction` depends upon the `direction` value against which it is checking?

Comment (Junchao-Mellanox) [reply src/system-health/health_checker/hardware_checker.py:144]: Hi, it is depending on the first fan's direction.

File: src/system-health/health_checker/hardware_checker.py (hunk 2 above)
Comment (Junchao-Mellanox) [reply src/system-health/health_checker/hardware_checker.py:85]: fixed

File: src/system-health/tests/test_system_health.py (hunk 1 above)
Comment (Junchao-Mellanox) [reply src/system-health/tests/test_system_health.py]: fixed

**Review by prgeor — APPROVED** [review]

**Review by liat-grozovik — APPROVED** [review]

**Comment by Junchao-Mellanox** [comment]
Hi @StormLiangMS , could you please help cherry-pick to 202211?

**Pull request merged**